src/backend/api/meals.js

```diff
@@ -5,11 +5,167 @@ const knex = require("../database");
+  }
+});
+
+router.get("/max-price", async (request, response) => {
+  try{
+    const maxPrice = parseFloat(request.query.maxPrice);
+    console.log(maxPrice);
+    if(isNaN(maxPrice)) {
+      response.status(400).json({ error: "maxPrice must be integers." });
+      return;
+    }
+    const mealsLessThanMaxPrice = await knex("meal").where('price', '<', maxPrice);
+    response.send(mealsLessThanMaxPrice);
+    return;
+  } catch (error) {
+    console.log(error.message);
+    response.status(500).send({ error: "Internal Server Error." });
+  }
+});
+
+router.get("/availableReservations", async (request, response) => {
+  try{
+      let availableReservations = request.query.availableReservations;
+      if(availableReservations = 'true') {
+        const mealsHasAvailableReservations = await knex("meal")
+          .select("meal.id", "meal.title", "meal.max_reservations")
+          .sum({ total_reserved: 'reservation.number_of_guests' })
+          .leftJoin("reservation", "meal.id", "reservation.meal_id")
+          .groupBy("meal.id")
+          .having("meal.max_reservations", ">", "total_reserved")
+        response.json(mealsHasAvailableReservations)
+        return;
+      }
+    
+  } catch (error) {
+    console.log(error.message);
+    response.status(500).send({ error: "Internal Server Error." });
+  }
+});
+
+router.get("/title", async (request, response) => {
+  try{
+    const title = request.query.title.toLowerCase();
+    console.log({title})
+    const mealWithTitle = await knex("meal")
+      .where("meal.title", "Like", `%${title}%`);
+    response.json(mealWithTitle);
+    return;
+    
+  } catch (error) {
+    response.status(500).send({ error: "Internal Server Error." });
+  }
+});
+
+router.get("/createdAfter", async (request, response) => {
+  try{
+    const createdAfter = new Date(request.query.createdAfter);
+    if (!createdAfter.getDate()) {
+      response.status(400).json({ error: "Date is not valid!" })
+    }
+
+    const mealsCreatedAfterDate = await knex("meal")
+    .where("created_date", ">=", createdAfter)
+    response.json(mealsCreatedAfterDate);
+    return;
+  } catch (error) {
+    console.log(error.message);
+    response.status(500).send({ error: "Internal Server Error." });
+  }
+});
+
+
+router.get("/limit", async (request, response) => {
+  try{
+    const numbersOfMeals = Number(request.query.limit);
+    if (isNaN(numbersOfMeals)) {
+      response.status(400).json({ error: "limit must be integers." });
+      return;
+    }
+    const limitedNumOfMeals = await knex("meal").limit(numbersOfMeals);
+    response.json(limitedNumOfMeals)
+    return;
+    
   } catch (error) {
-    throw error;
+    response.status(500).send({ error: "Internal Server Error." });
   }
 });
 
+
+
+//Adds a new meal
+
+router.post('/', async (request, response) => {
+  try {
+    let newMeal = request.body;
+    newMeal.when = new Date(newMeal.when);
+    newMeal.created_date = new Date(newMeal.created_date);
```

Comment:
Are these 3 lines used/needed?

Reply:
Yes, Firstly since my date columns are mandatory (not nullable) in datable, a date value should be provided. 
Secondly, date string from sent client (in request body as json) needs to converted into a date object otherwise Knex doesn't handle it itself and throws and error. Maybe there's a better workaround but I couldn't find one.

Reply:
I see now. You're actually changing the request body because `let newMeal = request.body` doesn't actually copy the object. So it should be the same if you just did `request.body.when = new Date(request.body.when)`.

Reply:
Yes, also using "newMeal" shortens the lines by replacing "request.body"